src/sentry/integrations/github/integration.py

```diff
@@ -227,23 +227,36 @@ def extract_source_path_from_source_url(self, repo: Repository, url: str) -> str
         _, _, source_path = url.partition("/")
         return source_path
 
-    def get_repositories(
-        self, query: str | None = None, fetch_max_pages: bool = False
-    ) -> Sequence[Mapping[str, Any]]:
+    def get_repositories(self, query: str | None = None, **kwargs: Any) -> list[dict[str, Any]]:
         """
-        This fetches all repositories accessible to a Github App
+        args:
+        * query - a query to filter the repositories by
+
+        kwargs:
+        * fetch_max_pages - fetch as many repos as possible using pagination (slow)
+
+        This fetches all repositories accessible to the Github App
         https://docs.github.com/en/rest/apps/installations#list-repositories-accessible-to-the-app-installation
 
-        per_page: The number of results per page (max 100; default 30).
+        It uses page_size from the base class to specify how many items per page (max 100; default 30).
+        The upper bound of requests is controlled with self.page_number_limit to prevent infinite requests.
         """
+        fetch_max_pages = kwargs.get("fetch_max_pages", False)
         if not query:
+            # XXX: In order to speed up this function we could use ThreadPoolExecutor
+            # to fetch repositories in parallel. See src/sentry/utils/snuba.py
+            repos = self.get_client().get_with_pagination(
+                "/installation/repositories",
+                response_key="repositories",
+                page_number_limit=self.page_number_limit if fetch_max_pages else 1,
+            )
             return [
                 {
                     "name": i["name"],
                     "identifier": i["full_name"],
                     "default_branch": i.get("default_branch"),
                 }
-                for i in self.get_client().get_repositories(fetch_max_pages)
```

Comment:
The added code all comes from the `get_repositories` I delete:
<img width="564" alt="image" src="https://github.com/user-attachments/assets/405f8ea0-8187-4839-a05b-8d995d85f440" />